Add friends likes

EugenePokh opened the pull request.

EugenePokh committed:
spring 9

EugenePokh committed:
spring 9

EugenePokh committed:
sprint 9 - fix dto validation for field duration

EugenePokh committed:
sprint 9

EugenePokh committed:
sprint 9 - fix Date to LocalDate (format of date)

EugenePokh committed:
sprint 9 - fix path of change of operations

EugenePokh committed:
sprint 9 - fix sevice logic

EugenePokh committed:
sprint 9 - fix sevice logic

EugenePokh committed:
sprint 9 - fix date of film for check

EugenePokh committed:
sprint 9 - fix date in tests

EugenePokh committed:
Refreshed after review

EugenePokh committed:
Sprint 9: refreshed after review + repsopnse (404 not found) + current date

EugenePokh committed:
Sprint 10: done

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,17 @@
+    private String description;
+    private LocalDate releaseDate;
+    private int duration;
+    private Set<Long> idForLikes = new HashSet<>();
```
лучше назвать userIdsForLikes, иначе не понятно, чей там id

EugenePokh replied:
Сделано, спасибо!

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/service/MostLikedComparator.java:
```diff
@@ -0,0 +1,15 @@
+package ru.yandex.practicum.filmorate.service;
```
лучше поместить в пакет service.impl, т. к. используется только в FilmServiceImpl

EugenePokh replied:
Сделано, спасибо!

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/FilmServiceImpl.java:
```diff
@@ -0,0 +1,61 @@
+
+    @Override
+    public void addLike(Film film, User user) {
+        film.getIdForLikes().add(user.getId());
```
очень подозрительно выглядит такой код, т. к. он ничего не сохраняет в filmStorage - если filmStorage будет работать с базой данных, такой код перестанет правильно работать

EugenePokh replied:
Сделано, спасибо!

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/FilmServiceImpl.java:
```diff
@@ -0,0 +1,61 @@
+
+    @Override
+    public void deleteLike(Film film, User user) {
+        film.getIdForLikes().remove(user.getId());
```
очень подозрительно выглядит такой код, т. к. он ничего не сохраняет в filmStorage - если filmStorage будет работать с базой данных, такой код перестанет правильно работать

EugenePokh replied:
Сделано, спасибо!

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    @Override
+    public void addFriend(User user, User friend) {
```
очень подозрительно выглядит такой код, т. к. он ничего не сохраняет в userStorage - если userStorage будет работать с базой данных, такой код перестанет правильно работать

EugenePokh replied:
Сделано, спасибо!

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    @Override
+    public void deleteFriend(User user, User friend) {
```
очень подозрительно выглядит такой код, т. к. он ничего не сохраняет в userStorage - если userStorage будет работать с базой данных, такой код перестанет правильно работать

EugenePokh replied:
Сделано, спасибо!

EugenePokh committed:
Sprint 10:
1. change name for likes in film
2. change folder of MostLikedComparator
3. add / delete like / friend for storage

EugenePokh committed:
Sprint 10: change save to update for like and friend services

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,83 @@
+    private final UserService userService;
+
+    @PostMapping("/films")
+    public Film createFilm(@Validated @RequestBody FilmRequestDto filmRequestDto) {
```
если контроллер принимает в методы DTO, то и возвращать должен тоже DTO, иначе теряется смысл DTO: передача данных между слоями и ничего больше – поэтому DTO лучше делать неизменяемыми (то есть их нельзя изменить после создания), в отличие от классов модели, которые могут быть изменяемыми
https://medium.com/nuances-of-programming/не-самые-очевидные-советы-по-написанию-dto-на-java-16ad8419d36b

EugenePokh replied:
Спасибо, сделано!

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/controller/dto/ErrorResponse.java:
```diff
@@ -0,0 +1,12 @@
+package ru.yandex.practicum.filmorate.controller.dto;
```
это не  DTO - лучше поместить внутрь BaseExceptionHandler, либо рядом с ним

EugenePokh replied:
Спасибо, сделано!

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/controller/dto/FilmRequestDto.java:
```diff
@@ -0,0 +1,26 @@
+
+@Data
+public class FilmRequestDto {
+    private Long id;
```
лучше сделать все поля final, чтобы DTO нельзя было изменить после создания

boristm1 commented on src/main/java/ru/yandex/practicum/filmorate/controller/dto/UserRequestDto.java:
```diff
@@ -0,0 +1,26 @@
+
+@Data
+public class UserRequestDto {
+    private Long id;
```
лучше сделать все поля final, чтобы DTO нельзя было изменить после создания
